easybuild/framework/easyblock.py

```diff
@@ -3110,11 +3110,23 @@ def print_post_install_messages(self):
             print_msg(msg, log=self.log)
 
     def post_install_step(self):
+        """[DEPRECATED] Do some postprocessing."""
+        # this is for easyblocks calling super(EB_xxx, self).post_install_step()
+        # deprecation warning is below, in post_processing_step
+        return self.post_processing_step()
+
+    def post_processing_step(self):
         """
         Do some postprocessing
         - run post install commands if any were specified
         """
 
+        if self.post_install_step.__qualname__ != "EasyBlock.post_install_step":
+            self.log.deprecated(
+                "EasyBlock.post_install_step() is deprecated, use EasyBlock.post_processing_step() instead.",
+                '6.0',
+            )
```

Comment:
@bartoldeman I had to remove the call to `self.post_install_step()` which you had here, otherwise we run into an infinite loop when using an easyblock that still defines `post_install_step` (see the enhanced version of `test_post_processing_step` which I've pushed in).

I don't think we need to call `post_install_step` here, and I also think we can/should move the deprecation warning into `post_install_step` above...

Any particular reason why you didn't take that approach?

Reply:
so this needs more work. I'll do this with another test.

Reply:
it was actually needed to run stuff inside the easyblock's `post_install_step()` (which wasn't tested) with or without the super. Super caused the recursion though.
Simplest solution is to keep `post_install_step` in `easyblock.py` unmodified from before the PR (!), then `post_processing_step` calls `post_install_step` unconditionally with or without deprecation warning.